### PR TITLE
[TextField] Improves styling on hidden ClearButton in TextField component.

### LINKED ISSUE #5888: [TextField] change properties on ClearButton in TextField to improve visibility in mobile
## Feature request summary
At mobile window sizes, certain `<TextField />`s with hidden `clearButton`s that are rendered in admin are blocking placeholder content. Examples can be seen here:
image
image

Proposed change:
in `/polaris-react/src/components/TextField/TextField.scss` there is a class called `.Hidden`. Instead of using the visibility property to hide the `clearButton`, we apply the `display: none` property.

## Rationale
image
Applying this new property will prevent placeholder text from being covered by an "invisible" white container.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by zaquille-oneil**
### WHY are these changes introduced?

Fixes #5888. Screenshots are shown in the github issue.

### WHAT is this pull request doing?
Renaming and adjusting a CSS style on the TextField component.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by ghost** [comment]
👋 Thanks for opening your first pull request. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris/blob/master/.github/CONTRIBUTING.md).

File: .changeset/strange-scissors-poke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Replacing .Hidden class with .hideClearButton in TextField.scss to address a UX issue caused in mobile views when the clear button in a TextField is not being shown.
```
Comment (chloerice) [inline .changeset/strange-scissors-poke.md]:
```suggestion
Fixed clear button covering placeholder text in `TextField` on small screens
```

File: polaris-react/src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -252,8 +252,8 @@ $spinner-icon-size: 12px;
   }
 }
 
-.Hidden {
-  visibility: hidden;
+.hideClearButton {
+  display: none;
 }
```
Comment (chloerice) [inline polaris-react/src/components/TextField/TextField.scss]:
```suggestion
```

File: polaris-react/src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -329,7 +329,7 @@ export function TextField({
 
   const clearButtonClassNames = classNames(
     styles.ClearButton,
-    !clearButtonVisible && styles.Hidden,
+    !clearButtonVisible && styles.hideClearButton,
```
Comment (chloerice) [inline polaris-react/src/components/TextField/TextField.tsx]:
```suggestion
```

File: polaris-react/src/components/TextField/TextField.tsx [hunk 4]
```diff
@@ -329,7 +329,7 @@ export function TextField({
...
   );
 
   const clearButtonMarkup = clearButton ? (
```
Comment (chloerice) [inline polaris-react/src/components/TextField/TextField.tsx]:
```suggestion
  const clearButtonMarkup = clearButton && clearButtonVisible ? (
```

**Review by chloerice — APPROVED** [review]
Thanks for catching and fixing this @zaquille-oneil!! 🎉 I think we can do away with the CSS altogether, but I'm having trouble running Storybook so I haven't tophatted my suggestion.

**Review by stefanlegg — APPROVED** [review]
LGTM! I'm fine with either the current approach or the one @chloerice suggested 👍

**Comment by zaquille-oneil** [comment]
> Thanks for catching and fixing this @zaquille-oneil!! 🎉 I think we can do away with the CSS altogether, but I'm having trouble running Storybook so I haven't tophatted my suggestion.

np! glad I could help. I've implemented your suggestion and ran storybook with it. things work just fine when hiding the clear button

**Pull request merged**